File: 030/030.go
package main

import (
	"fmt"
)

func main() {
	sum := 0
	for i := 355000; i > 2; i-- {
		j := i
		raisedSum := 0

		for j > 0 {
			raisedSum += pow((j % 10), 5)
			j /= 10
		}
		if i == raisedSum {
			sum += i
		}
	}
	fmt.Println(sum)
}

func pow(n, e int) int {
	s := 1
	for i := 0; i < e; i++ {
		s *= n
	}
	return s
}

File: solutions/030/030.c
#include <stdio.h>

int intPow(int n, int e) {
    int s = 1;
    for (int i = 0; i < e; i++)
        s *= n;
    return s;
}

int main() {
    int sum = 0;
    for (int i = 355000; i > 2; i--) {
        int j = i;
        int raisedSum = 0;

        while (j > 0) {
            raisedSum += intPow((j % 10), 5);
            j /= 10;
        }

        if (i == raisedSum)
            sum += i;
    }
    printf("%d\n", sum);
}
